app/models/message.rb

```diff
@@ -31,6 +31,25 @@ def create_from_queue(body)
     self
   end
 
+  def set_status!
+    successes = 0
+
+    transmissions = Transmission.where(:message_uuid => self.uuid)
+    transmissions.each do |t|
```

Comment:
If you want to do it like this. Rather than making a successes variable and counting them. I'd do 

```
successes = transmissions.select do |transmission|
  transmission.state == 'failed'
end
```

Then you can just do;

```
if successes.length == transmissions.length
```